packages/components/date-picker/src/date-picker-com/panel-date-pick.vue

```diff
@@ -361,12 +361,36 @@ type Shortcut = {
   onClick?: (ctx: Omit<SetupContext, 'expose'>) => void
 }
 
+const arrangeDatesShortcuts = (selectedValue: Dayjs) => {
+  if (!isArray(props.parsedValue)) {
+    return emit([selectedValue], true)
+  }
+  const valueOfParsedValue: string[] = (props.parsedValue as Dayjs[]).map(
+    (it) => dayjs(it).format('YYYY-MM-DD')
+  )
+
+  const isSelectedIndex = valueOfParsedValue.indexOf(
+    dayjs(selectedValue).format('YYYY-MM-DD')
+  )
+
+  emit(
+    isSelectedIndex > -1
+      ? castArray(props.parsedValue).filter((d, i) => i !== isSelectedIndex)
+      : (castArray(props.parsedValue) as Dayjs[]).concat([selectedValue]),
+    true
+  )
```

Comment:
The code here may be adjusted as follows:
```ts
const index = props.parsedValue.findIndex(date => days(date).format('YYYY-MM-DD') === dayjs(selectedValue).format('YYYY-MM-DD'))
const dates = index > -1 ? props.parsedValue.splice(index, 1) : props.parsedValue.push(selectedValue)
emit(dates, true)
```

Reply:
hello, I don't think the `parsedValue` of props should be changed inside the component, and the return value of `splice` or `push` is not expected. 
May I just refactor by extract `dates` like the following?
```
const dates = isSelectedIndex > -1
      ? castArray(props.parsedValue).filter((d, i) => i !== isSelectedIndex)
      : (castArray(props.parsedValue) as Dayjs[]).concat([selectedValue])
emit(dates, true)
```

Reply:
Oh, I made a mistake, really shouldn't modify `props.parsedValue` directly.

According to the code you provided, it seems that it can be simplified: 
```ts
const dates = isSelectedIndex > -1
      ? props.parsedValue.filter((d, i) => i !== isSelectedIndex)
      : (props.parsedValue as Dayjs[]).concat([selectedValue])
```

Reply:
done